File: 0455-assign-cookies/0455-assign-cookies.py
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        result = 0
        g.sort()
        s.sort()
        n_childen = len(g)
        n_cookies = len(s)
        for i in range(0, n_cookies):
            if result < n_childen and g[result] <= s[i]:
                result += 1
        return result

File: 0455-assign-cookies/0455-assign-cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int s_idx = s.size() - 1;
        int count = 0;
        for (int i = g.size() - 1; i >= 0; i--){
            if (s_idx >= 0 && g[i] <= s[s_idx]){
                count++;
                s_idx--;
            }
        }
        return count;
    }
};